### PR TITLE
Acceleration limitation + Angular velocity function of speed + cleanups

### PR DESCRIPTION
**Pull request by netgusto**

### CODE REVIEW

File: agents/attractor/attractor.go [hunk 1]
```diff
@@ -20,11 +20,22 @@ func MakeAttractorAgent() AttractorAgent {
 	}
 }
 
+var speedchange = 0.0
+var sign = 1.0
```
Comment (xtuc) [inline agents/attractor/attractor.go:24]: Are these golang constants?

File: agents/attractor/attractor.go [hunk 2]
```diff
@@ -20,11 +20,22 @@ func MakeAttractorAgent() AttractorAgent {
...
+
 func (agent AttractorAgent) SetPerception(perception state.Perception, comm protocol.AgentCommunicator, agentstate state.AgentState) {
 
 	speed := perception.Specs.MaxSpeed
 
-	desired := utils.MakeVector2(1, 20).SetMag(speed).Limit(perception.Specs.MaxSteeringForce)
+	if speedchange+0.01 >= speed {
```
Comment (xtuc) [inline agents/attractor/attractor.go:30]: The MaxSpeed spec given by the server can be negative?

I don't get this condition.

File: agents/attractor/attractor.go [hunk 3]
```diff
@@ -20,11 +20,22 @@ func MakeAttractorAgent() AttractorAgent {
...
+		sign = -1
+	} else if speedchange <= 0 {
+		sign = 1
+	}
+
+	speedchange += sign * 0.01
+
+	desired := utils.MakeVector2(1*sign, 20).SetMag(speed - speedchange)
```
Comment (xtuc) [inline agents/attractor/attractor.go:38]: Following my comment above. Do you use this to change the direction of the attractor?

File: server/state/agentstate.go [hunk 4]
```diff
@@ -68,14 +68,38 @@ func MakeAgentState() AgentState {
 }
 
 func (state AgentState) Update() AgentState {
+	if state.Velocity.Mag() > 0.00001 {
+		state.Orientation = state.Velocity.Angle()
+	} else {
+		state.Velocity = utils.MakeVector2(0, 0)
```
Comment (xtuc) [inline server/state/agentstate.go:74]: Could be nice to have a `reset` or `null` method as shorthand for this.

File: agents/attractor/attractor.go (hunk 1 above)
Comment (netgusto) [reply agents/attractor/attractor.go:24]: nope, these are just static variables (defined outside code).

File: server/state/agentstate.go (hunk 4 above)
Comment (netgusto) [reply server/state/agentstate.go:74]: Yes, we shall have a `MakeVector2Null()` method

**Comment by netgusto** [comment]
Closing; everything has been improved in branch feat-fov #43

**Pull request closed without merging**